UDV-Core/src/Extensions/DocumentLinks/views/DocumentLinkWindow.js

```diff
@@ -0,0 +1,409 @@
+          updateITownsView(this.itownsView, this.layer);
+          this.highlightedBuildingInfo = buildingInfo;
+        } catch (_) {
+          alert('Building is not currently in the view. Travel to it first');
+        }
+      } else {
+        alert('Building was not loaded by iTowns. Please navigate in the city' +
+          ' to make sure the building has been loaded at least once.');
+      }
+    }
+  }
+
+  /**
+   * If the target is a city object, moves the camera to focus on its centroid.
+   * 
+   * @param {string} type The link target type.
+   * @param {any} link The actual link, with `id`, `source_id` and `target_id`
+   */
+  async travelToLink(type, link) {
+    if (type === 'city_object') {
+      this.tbi = getTilesBuildingInfo(this.layer, this.tbi);
+      let buildingId = link.target_id;
+      let buildingInfo = this.tbi.buildings[buildingId];
+      if (!!buildingInfo) {
+        await focusCameraOn(this.itownsView, this.controls,
+          buildingInfo.centroid, {duration: 1});
+      } else {
+        alert('Building was not loaded by iTowns. Please navigate in the city' +
+          ' to make sure the building has been loaded at least once.');
+      }
+    }
+  }
+
+  /**
+   * Toggles the building selection.  
+   * When selecting a building, the mouse cursor position will be used to fetch
+   * building IDs under it, and a mouse click will trigger the selection of a
+   * building.
+   */
+  toggleBuildingSelection() {
+    this.isSelectingBuilding = !this.isSelectingBuilding;
+    if (this.isSelectingBuilding) {
+      if (this.isCreated) {
+        this.linkTablesDivElement.style.opacity = 0.5;
+        this.linkTablesDivElement.style.pointerEvents = 'none';
+        this.selectBuildingButtonElement.innerText = 'Cancel';
+        this.createLinkButtonElement.style.display = 'initial';
+      }
+      window.addEventListener('mousemove', this.mouseMoveListener);
+      window.addEventListener('mousedown', this.mouseClickListener);
+      this.clearHighlightedBuilding();
+    } else {
+      if (this.isCreated) {
+        this.linkTablesDivElement.style.opacity = 1;
+        this.linkTablesDivElement.style.pointerEvents = 'auto';
+        this.selectBuildingButtonElement.innerText = 'Select building';
+        this.createLinkButtonElement.style.display = 'none';
+        this.hoveredBuildingParagraphElement.innerText = 'No building';
+        this.selectedBuildingParagraphElement.innerText = '';
+      }
+      window.removeEventListener('mousemove', this.mouseMoveListener);
+      window.removeEventListener('mousedown', this.mouseClickListener);
+      this.clearSelectedBuilding();
+    }
+  }
+
+  /**
+   * If the user is currently hovering a building, fetches the building ID and
+   * displays it in the window.
+   * 
+   * @param {MouseEvent} event The mouse event.
+   */
+  onMouseMove(event) {
+    if (event.target.nodeName.toUpperCase() === 'CANVAS') {
```

Comment:
A comment explaining this if would ease understanding

Reply:
Done